app/src/org/commcare/android/database/app/AppDatabaseUpgrader.java

```diff
@@ -76,9 +77,13 @@ private boolean upgradeOneTwo(SQLiteDatabase db, int oldVersion, int newVersion)
     private  boolean upgradeThreeFour(SQLiteDatabase db) {
         db.beginTransaction();
         try {
-            db.execSQL("CREATE INDEX global_index_id ON GLOBAL_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
-            db.execSQL("CREATE INDEX upgrade_index_id ON UPGRADE_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
-            db.execSQL("CREATE INDEX recovery_index_id ON RECOVERY_RESOURCE_TABLE ( " + Resource.META_INDEX_PARENT_GUID + " )");
+            db.execSQL("CREATE INDEX global_index_id ON " + 
+                    CommCareApp.GLOBAL_STORAGE_TABLE_KEY +
+                    " ( " + Resource.META_INDEX_PARENT_GUID + " )");
+            db.execSQL("CREATE INDEX upgrade_index_id ON " + CommCareApp.UPGRADE_STORAGE_TABLE_KEY +
+                    " ( " + Resource.META_INDEX_PARENT_GUID + " )");
+            db.execSQL("CREATE INDEX recovery_index_id ON " + CommCareApp.RECOVERY_STORAGE_TABLE_KEY +
+                    " ( " + Resource.META_INDEX_PARENT_GUID + " )");
```

Comment:
Yes, I'd definitely hard-code any names during migration. Hm, that should actually be something we're doing for the other fields as well, good catch